packages/lib-user/src/components/Contributors/helpers/getAllUsers.js

```diff
@@ -0,0 +1,55 @@
+import { arrayOf, func, string } from 'prop-types'
+
+import { fetchPanoptesUsers } from '../../../utils'
+
+// maximum number of IDs per request, see GitHub issue front-end-monorepo/issues/6363
+// to keep request less than 2084 characters, with IDs ~6 characters long plus %2C2 (encoded comma), use chunk size of 200
+const CHUNK_SIZE = 200
+
+export async function getAllUsers({
+  memberIdsPerStats,
+  setExportProgress
+}) {
+  let allUsers = []
+
+  if (memberIdsPerStats && memberIdsPerStats.length > 0) {
+    // If there are more than CHUNK_SIZE users, make multiple requests
+    if (memberIdsPerStats.length > CHUNK_SIZE) {
+      const userPromises = []
+      const totalChunks = Math.ceil(memberIdsPerStats.length / CHUNK_SIZE)
+
+      for (let i = 0; i < memberIdsPerStats.length; i += CHUNK_SIZE) {
+        const chunk = memberIdsPerStats.slice(i, i + CHUNK_SIZE)
+        const chunkQuery = {
+          id: chunk.join(','),
+          page_size: 100 // per panoptes, maximum page_size is 100
+        }
+
+        const wrappedPromise = fetchPanoptesUsers(chunkQuery).then(result => {
+          setExportProgress(prevProgress => prevProgress + ((1 / totalChunks) * 100))
+          return result
+        })
+
+        userPromises.push(wrappedPromise)
```

Comment:
The wrapped promises are essentially the progress indicator. This is not exact. On very large groups the progress hangs around 90-98% as the very last few `/users` requests take awhile to return, and I'm not sure why. 

This makes me wonder - is a roughly accurate but not exact progress indicator that sometimes appears to but doesn't actually get stuck towards completion better or worse than a simple loading indicator? For most groups a progress status isn't necessary, but for the very large groups it feels like something is necessary.

Reply:
I checked out this branch locally, and confirmed I can generated an export for the large group and the small group. In the large group, I can replicate the loading state freeze at 98%, and I noticed that the loading animation also freezes at that point.

What would cause the `<Loader />` animation to stop? Is that a point in the export process that can be hooked into to display an extra message like "Finishing up..."? Because the animation also froze, I could easily assume the export failed.